logic.js

```diff
@@ -36,429 +36,439 @@ http://ricostacruz.com/cheatsheets/umdjs.html
     return a;
   }
 
```

Comment:
Most of the changes in this file are just due to indentation.  (Specifically: wrapping the existing logic w/ another function)